resource_obmcs_core_instance_test.go

```diff
@@ -154,34 +161,109 @@ func (s *ResourceCoreInstanceTestSuite) TestAccResourceCoreInstance_basic() {
 				}`,
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(s.ResourceName, "display_name", "-tf-instance"),
-					resource.TestCheckResourceAttr(vnicResourceName, "display_name", "-tf-vnic"),
-					resource.TestCheckResourceAttr(vnicResourceName, "skip_source_dest_check", "false"),
+					resource.TestCheckResourceAttr(s.ResourceName, "create_vnic_details.0.display_name", "-tf-vnic"),
+					resource.TestCheckResourceAttr(s.ResourceName, "create_vnic_details.0.skip_source_dest_check", "false"),
 					func(ts *terraform.State) (err error) {
 						newId, err := fromInstanceState(ts, s.ResourceName, "id")
-						if newId == instanceId {
-							return fmt.Errorf("Expected new instance ocid, got the same.")
+						if newId != instanceId {
+							return fmt.Errorf("Expected same instance ocid, got different.")
 						}
-						instanceId = newId
 						return err
 					},
 				),
 			},
-			// verify force new by creating an instance with different VNIC details.
+			// Adding create_vnic_details flags with default values should not lead to a change.
 			{
+				ImportState:       true,
+				ImportStateVerify: true,
 				Config: s.Config + `
 				resource "oci_core_instance" "t" {
 					availability_domain = "${data.oci_identity_availability_domains.ADs.availability_domains.0.name}"
 					compartment_id = "${var.compartment_id}"
 					image = "${data.oci_core_images.t.images.0.id}"
 					shape = "VM.Standard1.1"
 					display_name = "-tf-instance"
+					subnet_id = "${oci_core_subnet.t.id}"
```

Comment:
Does removing subnet_id here when adding the create_vnic_details object cause "force new"?

Reply:
It does, and that's the current behavior without this change as well.

That said, I think we can change this pretty easily. Right now we're not reading in subnet_id in SetData. If we add it there and then mark it as calculated, then I believe users can switch off of this without consequence. I think there's enough in this PR already though, so I have not done that.